segmentstore/server/src/main/java/io/pravega/segmentstore/server/logs/operations/MergeSegmentOperation.java

```diff
@@ -113,17 +127,38 @@ public long getLength() {
     @Override
     public String toString() {
         return String.format(
-                "%s, SourceSegmentId = %d, Length = %s, MergeOffset = %s",
+                "%s, SourceSegmentId = %d, Length = %s, MergeOffset = %s, Attributes = %d",
                 super.toString(),
                 getSourceSegmentId(),
                 toString(getLength(), -1),
-                toString(getStreamSegmentOffset(), -1));
+                toString(getStreamSegmentOffset(), -1),
+                this.attributeUpdates.size());
+    }
+
+    //endregion
+
+    //region AttributeUpdaterOperation implementation
+
+    @Override
+    public AttributeUpdateCollection getAttributeUpdates() {
+        return AttributeUpdateCollection.from(attributeUpdates);
+    }
+
+    @Override
+    public boolean hasOnlyCoreAttributes() {
```

Comment:
You may omit this method. It's already declared in the interface.

Reply:
Done.